subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/DependPlugin.java

```diff
@@ -132,6 +134,24 @@ public Boolean call() throws Exception {
                 return Clang.class.isAssignableFrom(toolChain.getClass()) || Gcc.class.isAssignableFrom(toolChain.getClass());
             }
         }));
+        depend.getToolChain().set(project.provider(new Callable<NativeToolChain>() {
+            @Override
+            public NativeToolChain call() throws Exception {
+                return compile.getToolChain();
+            }
+        }));
+        depend.getTargetPlatform().set(project.provider(new Callable<NativePlatform>() {
+            @Override
+            public NativePlatform call() throws Exception {
+                return compile.getTargetPlatform();
+            }
+        }));
+        depend.getSpec().set(project.provider(new Callable<NativeCompileSpec>() {
+            @Override
+            public NativeCompileSpec call() throws Exception {
+                return compile.createCompileSpec();
+            }
+        }));
```

Comment:
And my other idea, continuing from Slack:
- We create `Depend` directly in `CppBasePlugin` and wire the depend task in the same way we wire the compile task--directly from the `CppBinary`.  If someone changes the compile task without adjusting the depend task, that's just the way it'll be.  It's much easier to do it the right way with configurations now.
- We remove the use of the compiler or tool chain in `Depend`.  We should make the include path correct instead of trying to work it out from the tool chain.
- We will add compile-only dependencies on the "platform".  We can add these directly to the `cppCompileDebug` and `cppCompileRelease` configurations or wire them in through the `DefaultCppBinary.IncludePath` somehow.  To get the "platform" dependencies, we should lazily query the tool chain (add a `getSystemIncludes()` to `PlatformToolProvider` for now).

Reply:
Now the include path is correct and `Depend` does not know about the `PlatformToolProvider` any more.